tools/bin/go_core_tests

```diff
@@ -24,5 +24,7 @@ if [[ $EXITCODE != 0 ]]; then
   echo "Encountered test failures."
 else
   echo "All tests passed!"
+  # uploading coverage.txt to CodeCov
+  bash <(curl -s https://codecov.io/bash)
```

Comment:
Can we vendor this script instead of pulling it down every time? Maybe in `tools/bin/`.

Reply:
I was thinking about the same, but isn't it better to use the up-to-date version from codecov?

Reply:
This way we should always be compatible to codecov.

Reply:
Worth noting that there was a security issue with this a bit ago (see https://about.codecov.io/security-update/).

Reply:
We could use the pattern we are using on some generated code where we check in a copy, but also fetch again at runtime to verify it matches.

Reply:
We could use the pattern we are using on some generated code where we check in a copy, but also fetch again at runtime to verify it matches.

Reply:
Added codecov script